### DISCUSSION TITLE
Testing an InsertOne or Drop operation

### DISCUSSION THREAD

**Discussion by lu-jason** [sv-tools/mongoifc#82] [Q&A] [answered]
Hi,

Just wondering if it's possible to provide an example of how to use the mocks to test an InsertOne and/or a Drop operation on a collection?

Thanks

**Comment by SVilgelm** [comment]
Hi, have you read this discussion https://github.com/sv-tools/mongoifc/discussions/30 ?

It has an example of testing the Find function. The insert and delete are will be same.

I can provide a better example some time on this week

**Reply by lu-jason** [reply]
Yeah I think I figured it out, just needed the correct return type on the .Return for the collection. Extra examples might still be useful for developers using this for the first time, but no rush.

Thanks!

**Comment by SVilgelm** [comment, marked as answer]
I added some additional examples here https://github.com/sv-tools/mongoifc/blob/main/examples/simple/users_test.go

**Reply by SVilgelm** [reply]
and the collections: https://github.com/sv-tools/mongoifc/blob/main/examples/simple/collections_test.go

**Reply by lu-jason** [reply]
Looks good, thanks!
